### DISCUSSION TITLE
[SUGGESTION] Provide the owner/membership status of a private island.

### DISCUSSION THREAD

**Discussion by TGWaffles** [HypixelDev/ModAPI#31] [Ideas and Feedback]
Currently, as it was in locraw, the ClientboundLocationPacket simply shows "Private Island" when on a player's island, whether that is an island that the player is a part of, or someone else's island. Ideally I would like to write some behaviour that changes based on whether the island is the player's or someone else's, but the owner isn't trivial to detect, especially as longer names can be cut off in the tab list. At the moment, the best workaround is to detect the scoreboard text for "Your Island", but that doesn't provide the ownership information for other players' islands, nor does it work on garden - all gardens show as "The Garden".

The ideal solution would be, for example, a list of all of the coop members on the current private island, but as I'm unsure where that would fit best, a field denoting the profile id or whether the player is a part of the island would be a great substitute. The same field would also apply to the garden.

**Comment by mdashlw** [comment]
Imo it shouldn't (and maybe isn't possible to) be in ClientboundLocationPacket. Skyblock events are a whole other can of worms, there could be a packet for anything and everything. Though island info would be a good starting point.

**Comment by ItsEmpa** [comment]
best way i have found to combat this is to check the next scoreboard title update after the ClientboundLocationPacket. If the title has "GUEST", then you are on another person's island.

**Reply by TGWaffles** [reply]
I mentioned this, or well the inverse (checking the scoreboard for "Your Island"), but as I mentioned in the original suggestion, it doesn't provide any information as to **whose** island you're on - just that it's not yours. 

It could even be your own island, just on another profile.

**Reply by ItsEmpa** [reply]
true fair enough

also i dont think you can visit your own island on a different profile?
